.pre-commit-config.yaml

```diff
@@ -1,16 +1,43 @@
 repos:
--   repo: https://github.com/ambv/black
-    rev: 19.3b0
-    hooks:
-    - id: black
-      language_version: python3.7
 -   repo: https://github.com/pre-commit/pre-commit-hooks
-    rev: v2.1.0
+    rev: v2.4.0
     hooks:
-    -   id: end-of-file-fixer
     -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-docstring-first
+    -   id: check-yaml
+    -   id: debug-statements
+    -   id: check-ast
 -   repo: https://gitlab.com/pycqa/flake8
-    rev: 3.7.8
+    rev: 3.7.9
     hooks:
     -   id: flake8
-        args: ['--max-line-length=500', '--ignore=E203,E266,E501,W503', '--max-complexity=18', '--select=B,C,E,F,W,T4,B9']
+        args:
+          - --max-line-length=500
+          - --ignore=E203,E266,E501,W503
+          - --max-complexity=18
+          - --select=B,C,E,F,W,T4,B9
+-   repo: https://github.com/ambv/black
+    rev: 19.10b0
+    hooks:
+    - id: black
+      language_version: python3.7
+-   repo: https://github.com/asottile/pyupgrade
+    rev: v1.25.2
+    hooks:
+    -   id: pyupgrade
+        args: ['--py36-plus']
```

Comment:
Can you either doctor the commit message or add a comment about what each of these things does? e.g. I don't know what `pyupgrade` is about

Reply:
See my updated comment [here](https://github.com/python-adaptive/adaptive/pull/237#issue-351901333)

Reply:
and https://github.com/python-adaptive/adaptive/pull/237/commits/0a5cbe1d40a6b11a99c1b5262b411fc43d936db4